app/scripts/modules/core/delivery/filter/executionFilter.service.ts

```diff
@@ -11,36 +12,34 @@ export class ExecutionFilterService {
   private lastApplication: Application = null;
   private isFilterable: (sortFilterModel: any[]) => boolean;
 
-  // this gets called every time the URL changes, so we debounce it a tiny bit
-  public updateExecutionGroups: (application: Application) => IExecutionGroup[];
-
   constructor(private executionFilterModel: ExecutionFilterModel,
               private timeBoundaries: any,
               private $log: ILogService,
               private filterModelService: any,
               private pipelineConfig: any) {
     'ngInject';
     this.isFilterable = filterModelService.isFilterable;
+  }
 
-    this.updateExecutionGroups = debounce((application: Application) => {
-      if (!application) {
-        application = this.lastApplication;
-        if (!this.lastApplication) {
-          return null;
-        }
+  @Debounce(25)
+  public updateExecutionGroups(application: Application): IExecutionGroup[] {
```

Comment:
The return value is just there to simplify testing.

This is not something we want to memoize, as its main purpose is to mutate the data inside the execution filter model.